### DISCUSSION TITLE
`vsmarketplace@microsoft.com` won't tell me why my extension was locked / blocked so I can fix it, possible political censorship?

### DISCUSSION THREAD

**Discussion by cirosantilli** [microsoft/vscode-discussions#1824] [Extension Development QnA] [answered]
Hello,

I'm the author of the following extension: https://marketplace.visualstudio.com/items?itemName=ourbigbook.ourbigbook-vscode which has been marked as locked / blocked and cannot be installed anymore. On vs code it is marked as:

> This extension was reported to be problematic 

![Screenshot from 2024-09-18 16-56-29](https://github.com/user-attachments/assets/3b34e256-4f43-4a61-b496-4c162f3ca133)

I contacted `vsmarketplace@microsoft.com` and they replied:

> We have locked the extension after a thorough investigation due to security and privacy concerns.

but I'm not sure what is the problem and they have gone silent to my request for further clarification with a more precise reason so that I might be able to fix it and get the extension back up:

> Is there any way to find out why it was marked as problematic so I can solve the issue and get it published again?

Could anyone take a look?

I have a suspicion.

My extension adds language feature for a markup language I created: https://github.com/ourbigbook/ourbigbook and the extension source in particular is at: https://github.com/ourbigbook/ourbigbook/tree/master/vscode

In all of my software projects, I add a bit of anti Chinese government propaganda, as a way to test the freedom of speech of various hosts. In this case, the extension depends on the main markup implementation:

https://www.npmjs.com/package/ourbigbook

which in turn depends on the following propaganda package:

https://www.npmjs.com/package/china-dictatorship

which basically contains a dump of anti-CCP text present at: https://github.com/cirosantilli/china-dictatorship

The usage of this propaganda is extremely minor, and could be removed if necessary.

The main goal of that package is not the political propaganda. I've spent 3 weeks of my life coding it. It has value. I will remove and comply if needed. The political propaganda just went in without me thinking too much as in any of my other projects.

I'm not collecting any user data or disrupting computers in any way. It's just a dump of anti-Chinese government text.

Furthermore, there is nothing propaganda-specific to the vs code extension, it all just comes from my package itself.

But I need to know what is the problem so I can fix it. It would be too risky to just try and guess that this is the problem, to re-upload a new version of the package, and then just have my entire account taken down because of uploading multiple malicious packages.

If political propaganda is forbidden as a dependency of any vs code extension, then fine, state that clearly, and I will comply.

But as things stand, it is possible that marketplace admins want to forbid political propaganda, but also don't want to admit it, because they know this could look bad from a freedom of speech point of view.

But this kind of lack of transparency only makes things worse in the log run, as it fuels speculation and reduces confidence in the platform.

Sorry if the problem has nothing to do with this, but it's the only thing I can think of, and I'd rather be completely transparent about everything I do.

Thank you enormously,

Ciro Santilli from the OurBigBook Project

Cc. @burkeholland and @isidorn because I've seen you reply in other threads and you seem to really care about the users.

**Comment by isidorn** [comment, marked as answer]
We removed it by mistake - AI scanner flagged it as malicious. It was not removed due to anything related to politics. I see Marketplace already brought the extension back https://marketplace.visualstudio.com/items?itemName=ourbigbook.ourbigbook-vscode
And we will also bring it back in VS Code.
Thanks for reaching out.

**Reply by cirosantilli** [reply]
Thank you so much for the reply!

When I look on vs code now, it is still marked as problematic, and install button grayed out:

![Screenshot from 2024-09-18 16-56-29](https://github.com/user-attachments/assets/3b34e256-4f43-4a61-b496-4c162f3ca133)

Is it something that I might have cached locally?

First I  had noticed the problematic thing locally. Only later was https://marketplace.visualstudio.com/items?itemName=ourbigbook.ourbigbook-vscode down. And then the site was back up this morning, but problematic thing still show. So maybe two separate issues in play?

**Reply by sketchbuch** [reply]
In the marketplace it looks fine to me, but in VSCode it shows as problematic. I guess that is what @isidorn meant by fixing it in vscode too

**Reply by cirosantilli** [reply]
Oops, reading is hard. Thank you! :heart: :heart: :heart: :heart: :heart: :heart:
